articles/graph-data-structure-python-part1/index.md

```diff
@@ -0,0 +1,238 @@
+
+- [Graphs Based on Direction](#based-on-direction)
+- [Graphs Based on Weights](#based-on-weights)
+- [Special Graphs](#special-graphs)
+
+#### Based on Direction
+##### Undirected Graphs
+In an undirected graph, the edges have no path or direction. If there is a path from vertex X to vertex Y, then there is a path from vertex Y to vertex X. Edge (X, Y) represents the edge connecting vertex X to vertex Y.
+
+That is, `edge (X, Y) == edge (Y, X)`.
+
+![Undirected Graph](/engineering-education/graph-data-structure-python-part1/undirected-graph.png)
+
+*Figure: [Undirected Graph](https://pediaa.com/what-is-the-difference-between-directed-and-undirected-graph/)*
+
+##### Directed Graphs
+In a directed graph or digraph, the edges have an orientation. If there is a path from vertex X to vertex Y, then there isn't necessarily a path from vertex Y to vertex X.
+
+That is, `edge (X, Y) != edge (Y, X)`.
+
+![Directed Graphs](/engineering-education/graph-data-structure-python-part1/directed-graph.png)
+
+*Figure: [Directed Graph](https://pediaa.com/what-is-the-difference-between-directed-and-undirected-graph/)*
+
+#### Based on Weights
+##### Weighted Graphs
+A weighted graph has a value associated with every edge. The value may represent quantities like cost, distance, time, etc., depending on the graph. An edge of a weighted graph is represented as, `(u, v, w)`.
+
+- `u` -> Source vertex
+- `v` -> Destination vertex
+- `w` -> Weight associated to go from u to v.
+
+![Weighted Graphs](/engineering-education/graph-data-structure-python-part1/weighted-graph.png)
+
+*Figure: [Weighted Graph](https://www.clipartmax.com/middle/m2i8Z5i8G6b1b1b1_directed-weighted-ex1-directed-weighted-graph/)*
+
+##### Unweighted Graphs
+An unweighted graph does not have a value associated with every edge. An edge of an unweighted graph is represented as, `(u, v)`.
+
+- `u` -> Source vertex
+- `v` -> Destination vertex
+
+![Unweighted Graphs](/engineering-education/graph-data-structure-python-part1/unweighted-graph.png)
+
+*Figure:[Unweighted Graph](https://www.researchgate.net/figure/Unweighted-Undirected-Graph_fig3_327864836)*
+
+#### Special Graphs
+##### Trees
+An undirected graph with zero cycles is called a tree. A cycle in a graph is a sequence with the first and last vertices in the repeating sequence.
+
+It has X vertices and X-1 edges.
+
+![Tree - Graphs](/engineering-education/graph-data-structure-python-part1/tree-graph.png)
+
+*Figure:[Tree](https://en.wikipedia.org/wiki/Tree_(graph_theory))*
+
+##### Rooted Tree
+A rooted tree is a tree that has a designated root node. If edges point away from the root, it is called an **arborescence/out-tree**. If edges point towards the root, it is called an **anti-arborescence/in-tree**.
+
+![Rooted Tree](/engineering-education/graph-data-structure-python-part1/rooted-trees.png)
+
+*Figure: Rooted Tree*
```

Comment:
just wanted to double-check if this image needed a source

Reply:
@hectorkambow I saw the figure in a YouTube video. Added the link to the video as a source.